### PR TITLE
chore: enable public_api_docs enforcement.

### LINKED ISSUE #1302: Re-enable `public_member_api_docs` analysis option for `shorebird_cli`
There have been a couple occasions where I've been in an area of the code I didn't write and was not 100% clear about the purpose of the class/method in question. Enforcing the presence of doc strings would help with this.

### PR DESCRIPTION
**Pull request by eseidel**
I disabled it on a per-file basis in a few places, but turned it
on by default.

Also added cspell:words to some files that I touched.

Closes https://github.com/shorebirdtech/shorebird/issues/1302

### CODE REVIEW

File: packages/shorebird_cli/lib/src/auth/auth.dart [hunk 1]
```diff
@@ -19,10 +22,10 @@ import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
 
 export 'ci_token.dart';
 
-// A reference to a [Auth] instance.
+/// A reference to a [Auth] instance.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/auth/auth.dart]:
```suggestion
/// A reference to an [Auth] instance.
```

File: packages/shorebird_cli/lib/src/commands/upgrade_command.dart [hunk 2]
```diff
@@ -16,6 +16,7 @@ class UpgradeCommand extends ShorebirdCommand {
   @override
   String get description => 'Upgrade your copy of Shorebird.';
 
+  /// Name of the command, exposed for the CommandRunner.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/upgrade_command.dart]:
```suggestion
  /// Name of the command, exposed for the [CommandRunner].
```

File: packages/shorebird_cli/lib/src/artifact_manager.dart [hunk 3]
```diff
@@ -1,3 +1,4 @@
+// cspell:words archs xcarchive xcframework
```
Comment (erickzanardo) [inline packages/shorebird_cli/lib/src/artifact_manager.dart:1]: Maybe this should be configured globally on cspell?

File: packages/shorebird_cli/lib/src/config/shorebird_yaml.dart [hunk 4]
```diff
@@ -48,7 +49,9 @@ class ShorebirdYaml {
   final bool? autoUpdate;
 }
 
+/// Extension on Shorebird yaml to get the app id for a specific flavor.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/config/shorebird_yaml.dart]:
```suggestion
/// Extension on [ShorebirdYaml] to get the app id for a specific flavor.
```

File: packages/shorebird_cli/lib/src/engine_config.dart [hunk 5]
```diff
@@ -1,9 +1,11 @@
+// ignore_for_file: public_member_api_docs
+
 import 'package:scoped_deps/scoped_deps.dart';
 
-// A reference to a [EngineConfig] instance.
+/// A reference to a [EngineConfig] instance.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/engine_config.dart]:
```suggestion
/// A reference to an [EngineConfig] instance.
```

File: packages/shorebird_cli/lib/src/executables/adb.dart [hunk 6]
```diff
@@ -60,6 +60,7 @@ class Adb {
     }
   }
 
+  /// Runs adb logcat.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/executables/adb.dart]:
```suggestion
  /// Runs `adb logcat`.
```

File: packages/shorebird_cli/lib/src/executables/shorebird_tools.dart [hunk 7]
```diff
@@ -37,6 +38,7 @@ class ShorebirdTools {
     return shorebirdToolsDirectory.existsSync();
   }
 
+  /// The directory containing the shorebird_tools package.
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/executables/shorebird_tools.dart]:
```suggestion
  /// The directory containing the `shorebird_tools` package.
```

File: packages/shorebird_cli/lib/src/extensions/shorebird_process_result.dart [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import 'package:collection/collection.dart';
 import 'package:shorebird_cli/src/shorebird_process.dart';
 
+/// Extensions for finding app.dill in a [ShorebirdProcessResult].
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/extensions/shorebird_process_result.dart]:
```suggestion
/// Extensions for finding `app.dill` in a [ShorebirdProcessResult].
```

File: packages/shorebird_cli/lib/src/artifact_manager.dart (hunk 3 above)
Comment (eseidel) [reply packages/shorebird_cli/lib/src/artifact_manager.dart:1]: Yeah, we need to figure out our cspell settings.  Some of these definitely belong global.

File: packages/shorebird_cli/lib/src/shorebird_process.dart [hunk 9]
```diff
@@ -1,3 +1,5 @@
+// ignore_for_file: public_member_api_docs
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/shorebird_process.dart]: Why are we ignoring for this file? Looks like you added docs.

**Review by felangel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/shorebird_process.dart (hunk 9 above)
Comment (eseidel) [reply packages/shorebird_cli/lib/src/shorebird_process.dart]: Fixed.

**Pull request merged**